core/templates/calendar.html

```diff
@@ -1,29 +1,98 @@
+			<div id="weekView">
+				<button id="backBtn" disabled=true>Back</button><button id="forwardBtn">Forward</button>
+				<br>
+				<table id="weekGrid">
+					<% var days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]; %>
+					<% for (var i=7; i<=20; i++) { if (i == 7) { %>
+						<tr>
+							<th></th>
+								<% for (var k=0; k<days.length; k++) { %>
+								<th class="day" id="<%= new Date(date.getTime()+((k)-(date.getDay()-1))*86400000).toDateString() %>">
+									<%= days[k]+" "+new Date(date.getTime()+((k)-(date.getDay()-1))*86400000).getDate() %>
+								</th>
+							<% } %>
+						</tr>
+				<% } else { %>
+					<tr>
+						<% for (var j=0; j<=6; j++) { if (j==0) { %>
+							<th class="time"><%= (i!=7)?((i<=12)?i:i-12):"" %>:00</th>
+						<% } else { %>
+							<td id= "<%=days[j-1]+" "+i%>" data-time="<%=i%>" data-day="<%=j%>" ></td>
+						<% } } %>
+					</tr>
+				<% } } %>
+				</table>
+			</div>
+			<div id="dayView" class="hidden">
+				<button id="date"><%= date.toDateString() %></button>
+				<button id="previousDay" disabled=true>Previous Day</button><button id="nextDay">Next Day</button>
+				<table id="dayGrid">
+					<% var rooms = ["Recording Studio", "1141A", "1141B", "1141C", "1141D"]; %>
+					<% for (var i=7; i<=20; i++) { if (i == 7) { %>
+						<tr>
+							<th></th>
+								<% for (var k=0; k<rooms.length; k++) { %>
+								<th class="room">
+									<%= rooms[k] %>
+								</th>
+							<% } %>
+						</tr>
+				<% } else { %>
+					<tr>
+						<% for (var j=0; j<=rooms.length; j++) { if (j==0) { %>
+							<th class="time"><%= (i!=7)?((i<=12)?i:i-12):"" %>:00</th>
+						<% } else { %>
+							<td id= "<%=rooms[j-1]+" "+i%>" data-time="<%=i%>" data-room="<%= rooms[j-1] %>"></td>
+						<% } } %>
+					</tr>
+				<% } } %>
+				</table>
+			</div>
+			<div id="popup" style="position: absolute;" class="hidden">
+				Room: <select id="roomSelect">
+					<option>Recording Studio</option>
+					<option>1141A</option>
+					<option>1141B</option>
+					<option>1141C</option>
+					<option>1141D</option>
+				</select><br>
+				Date: <select id="dateSelect">
+					<% for (var i = 0; i <= 14; i++) { %>
+						<% var newDate =  new Date(date.getTime()+86400000*i) %>
+						<% if (newDate.getHours() >= 21) { %>
+						  <% newDate.setTime(newDate.getTime() + 86400000); %>
+						<% } %>
+						<% if (newDate.getDay() !== 0) { %>
+							<option> <%= newDate.toDateString() %> </option>
+						<% } %>
+					<% } %>
+				</select><br>
+				Time: <select id="timeSelect">
+					<% for (var i = 8; i<=20; i++) { %>
+						<option value=<%= i %>><%= (i!=7)?((i<=12)?i:i-12):"" %>:00</option>
+					<% } %>
+				</select><br>
+				Duration: <select id="durationSelect">
+					<option value="1">1 Hour</option>
+					<option value="2">2 Hours</option>
+					<option value="3">3 Hours</option>
+				</select><br>
+				Event Name: <input id=nameInput></input><br>

```

Comment:
A `maxlength` attribute here will ensure that the title here doesn't exceed that of the database spec